### ISSUE TITLE
Calibration failed, check configuration

### ISSUE THREAD

**Issue by Burbacottero** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#199]
### Checklist

- [X] I have updated the card to the latest version available
- [X] I have cleared the cache of my browser
- [X] I have checked if the problem is already reported

### The problem

When I start cleaning zones it tells calibration not valid check configuration.

![image](https://user-images.githubusercontent.com/82754488/145584718-ccec3740-9810-49e3-8afc-1551d4ec6de6.png)

I'm using the configuration from demo, with xiaomi cloud map extractor, configured like this>
```
camera:
  - platform: xiaomi_cloud_map_extractor
    scan_interval: 1
    host: 192.168.1.47
    token: !secret token
    username: !secret username
    password: !secret xiaomi_cloud_password
    draw: ["all"]
    colors:
      color_map_outside: [0, 0, 0]
      color_path: [255, 255, 255]
    map_transformation:
      rotate: 180
      trim:
        left: 25
        right: 4
        top: 25
        bottom: 22
    attributes:
      - calibration_points
```

### What version of a card has described problem?

v2.0.8

### What vacuum model do you have problems with?

roborock.vacuum.s5e

### Which integration do you use to control your vacuum (link)?

https://www.home-assistant.io/integrations/xiaomi_miio/

### What browser (browsers/apps) does have this problem?

Chrome

### What version of Home Assistant do you use?

core-2021.11.5

### What type of installation are you running?

Home Assistant OS

### Card's configuration

```yaml
type: custom:xiaomi-vacuum-map-card
title: ''
preset_name: Live map
entity: vacuum.chang
language: it
map_source:
  camera: camera.xiaomi_cloud_map_extractor
calibration_source:
  camera: true
map_modes:
  - template: vacuum_clean_zone
  - template: vacuum_clean_segment
    predefined_selections:
      - id: 1
        outline:
          - - 25362
            - 29989
          - - 25362
            - 30370
          - - 25421
            - 30370
          - - 25421
            - 33476
          - - 25333
            - 33476
          - - 25333
            - 33534
          - - 25274
            - 33534
          - - 25274
            - 33739
          - - 24073
            - 33710
          - - 24073
            - 33622
          - - 22520
            - 33622
          - - 22520
            - 33388
          - - 21963
            - 33388
          - - 21963
            - 33329
          - - 21875
            - 33329
          - - 21875
            - 33270
          - - 21641
            - 33270
          - - 21641
            - 33183
          - - 21524
            - 33183
          - - 21524
            - 33124
          - - 21465
            - 33124
          - - 21465
            - 32772
          - - 21670
            - 32772
          - - 21670
            - 31220
          - - 21524
            - 31220
          - - 21524
            - 30663
          - - 24278
            - 30663
          - - 24278
            - 29989
          - - 25362
            - 29989
        label:
          text: Bedroom
          x: 22800
          'y': 31000
          offset_y: 35
        icon:
          name: mdi:bed
          x: 22800
          'y': 31250
  - template: vacuum_goto
  - template: vacuum_follow_path
map_locked: false
two_finger_pan: true
vacuum_platform: default
```

### Javascript errors shown in the browser's console (if applicable)

```shell
Nothing
```

**Comment by Burbacottero** [comment]
> Ok, this is not a problem with card, but with map extractor: map disappears for a moment when vacuum starts cleaning

the map doesn't disappears😅, so what do I need to do?

**Comment by PiotrMachowski** [comment]
Not map card, but map in camera itself

**Comment by Burbacottero** [comment]
I don't understand, what I gave you it's the map itself

**Comment by Burbacottero** [comment]
then?

**Comment by philder** [comment]
> Ok, this is not a problem with card, but with map extractor: map disappears for a moment when vacuum starts cleaning

How to solve this?

**Comment by PiotrMachowski** [comment]
You can't, just wait for next release

**Comment by VadimKudrin** [comment]
Same issue. Another map_mode is working, for example vacuum_follow_path - I can point the vacuum to the place on  map and it succefull reach it. But when I run the room cleaning, the map disappears with error on the lovelace card "Calibration failed, check configuraiton" until the vcuum returned to the docking station.

**Comment by PiotrMachowski** [comment]
@VadimKudrin but does it clean correct room?

**Comment by ddmmcz** [comment]
It's the way others write. vacuum_clean_zone_predefined works normally, but vacuum_clean_segment flashes with the error "Calibration failed, check configuration" and starts cleaning immediately and does not arrive at the cleaning point immediately.

**Comment by PiotrMachowski** [comment]
@ddmmcz have you provided a correct room ID?

**Comment by ddmmcz** [comment]
Well I thought the ID is just for internal identification, or did I misunderstand it and does it have to fit the app somehow?

**Comment by PiotrMachowski** [comment]
@ddmmcz it has to match room number from the vacuum

**Comment by ddmmcz** [comment]
@PiotrMachowski I'm sorry I'm a little stubborn, but now I don't know exactly where to find it, thank you.

**Comment by ddmmcz** [comment]
@PiotrMachowski I found where to find the ID, so I'll test everything again.

**Comment by PiotrMachowski** [comment]
[How to find room ID](https://community.home-assistant.io/t/xiaomi-vacuum-interactive-map-card/123901/887)

**Comment by noorbertt** [comment]
I have a similar issue:

![image](https://user-images.githubusercontent.com/25225562/147847628-2012f008-7d92-424d-9fb3-54d84d8506a3.png)

![image](https://user-images.githubusercontent.com/25225562/147847617-189606da-a11c-492f-9f91-27eba09538ca.png)

![image](https://user-images.githubusercontent.com/25225562/147847634-6c33c25e-ecd2-4bb2-958a-a599091952c3.png)

`type: custom:xiaomi-vacuum-map-card
map_source:
  camera: camera.roborock_s5
  crop:
    top: 60
    bottom: 65
    left: 75
    right: 45
calibration_source:
  camera: true
entity: vacuum.roborock_vacuum_s5
vacuum_platform: default
map_locked: true
two_finger_pan: false
map_modes:
  - template: vacuum_clean_zone
  - template: vacuum_clean_zone_predefined
    predefined_selections:
      - zones:
          - - 25035
            - 18354
            - 29935
            - 23004
          - - 25051
            - 22895
            - 27701
            - 24695
        icon:
          name: mdi:sofa
          x: 27200
          'y': 21150
      - zones:
          - - 27673
            - 23077
            - 30023
            - 24577
          - - 28539
            - 24586
            - 29839
            - 26936
        icon:
          name: mdi:door
          x: 28950
          'y': 24150
      - zones:
          - - 25587
            - 24678
            - 27237
            - 26928
        icon:
          name: mdi:pot-mix
          x: 26387
          'y': 25758
  - template: vacuum_goto
  - template: vacuum_goto_predefined
    predefined_selections:
      - position:
          - 26400
          - 26550
        icon:
          name: mdi:broom
          x: 26400
          'y': 26550
`

`
- platform: xiaomi_cloud_map_extractor
  host: !secret roborock_s5_host
  token: !secret roborock_s5_token
  username: !secret miaccount_user
  password: !secret miaccount_pass
  name: "Roborock S5"
  store_map_path: "/config/www/vacuum/"
  store_map_image: true
  draw: ['all']
  map_transformation:
    scale: 2
    rotate: 90
    trim:
      top: 20
      bottom: 20
      left: 25
      right: 25
  attributes:
    - calibration_points
 `

It was fine an day ago

**Comment by horve000** [comment]
Same problem here.
![Schermafbeelding 2022-01-01 134457](https://user-images.githubusercontent.com/6498608/147850843-7ef72350-5510-422f-b984-715114719f4d.png)

All worked fine till i did upgrade my Roborock S5 to the newest firmware update. Now version 3.5.8_1566

**Comment by rschuiling** [comment]
Same issue for me 'invalid calibration' and it also seems to relate to upgrading the firmware of the S5 Max to 3.5.8_1566.

**Comment by raptr55** [comment]
Other custom components are failing:
https://github.com/al-one/hass-xiaomi-miot/issues/345

Seems like a problem in MI servers but not CN one

**Comment by ricarva** [comment]
Just to chime in with the same problem, and to second @raptr55 with the assessment that something seems to be off with the Xiaomi servers.

**Comment by minkola** [comment]
I'm getting the same error, cannot see my map says "Failed to Retrieve Devices"

I can still control the devices via HA but cannot view the camera though HA

Did work a day or 2 ago, have restored HA and still same issue, have removed integration and re-added but still same issue, also cannot generate a API key using "python3 token_extractor.py"

Logging in...
Logged in.

No devices found for server "cn".
Unable to get devices.
No devices found for server "us".
No devices found for server "ru".
No devices found for server "tw".
No devices found for server "sg".
No devices found for server "in".
No devices found for server "i2".

I have a S5 and S5 Max and I remember they were on the de server which is not in the list any more

**Comment by zollek** [comment]
I've the same problem with Roborock S5 (firmware 3.5.8_002034), don't remember doing any update.
The problem is not related to the official integration.

There is something wrong with extracting map from the DE server:
`access_token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`
`calibration_points: null`
`charger: null`
`cleaned_rooms: null`
`country: null`
`goto: null`
`map_name: null`
`no_go_areas: null`
`no_mopping_areas: null`
`room_numbers: []`
`rooms: null`
`vacuum_position: null`
`vacuum_room: null`
`vacuum_room_name: null`
`entity_picture: >-`
  `/api/camera_proxy/camera.xiaomi_cloud_map_extractor?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`
`friendly_name: Xiaomi Cloud Map Extractor`
`supported_features: 1`

**Comment by PiotrMachowski** [comment]
This issue should be fixed in v2.2.0
